api/src/main/java/io/opencensus/trace/export/SampledSpanStore.java

```diff
@@ -370,4 +384,61 @@ public static ErrorFilter create(
      */
     public abstract int getMaxSpansToReturn();
   }
+
+  @ThreadSafe
+  private static final class NoopSampledSpanStore extends SampledSpanStore {
```

Comment:
This is not a no-op implementation. @sebright you did the same for ViewManager but I think it is wrong to say that this is a no-op implementation. Based on my knowledge no-op means "do nothing".

Reply:
@bogdandrutu Do you object to just the name, or also the behavior?  I think it makes sense to rename the "Noop" classes that have state.

Reply:
@bogdandrutu I do agree the name is misleading, but I kind of understand why @sebright introduce such behavior at first. 

Unlike the get() apis, whose results rely on actual metrics we collected, the behaviors of these register/unregister apis are deterministic no matter we collect metrics or not. I think some users might have tests or condition checks on these apis?

Reply:
@sebright @HailongWen I don't like that we have a state (at all) in the no-op implementation in our API. But we should move forward with this PR and probably decide later what to do.